### DISCUSSION TITLE
What is the difference between `sum_of_diracs` and `with_peaks_at`?

### DISCUSSION THREAD

**Discussion by ipcamit** [e3nn/e3nn#346] [Q&A] [answered]
If generate spherical function coefficients using 
```python
s = e3nn.io.SphericalTensor(5,1,1)
pos = torch.tensor([
    [1.0, 1.0, 0.0],
    [0.0, 1.0, 0.0],
])

val = torch.tensor([
    -5.0,1.0
])
x1 = s.sum_of_diracs(pos,val)
x2 = s.with_peaks_at(pos,val)
```
What is the difference between the two approaches? When I plot the results, they are similar but not numerically identical.

**Comment by mariogeiger** [comment, marked as answer]
`sum_of_diracs` sums approximate dirac functions. If L was set to infinity the two functions would be identical. 
`with_peaks_at` is more "clever" than `sum_of_diracs` because it ensures that the evaluation at `pos` gives you `val` for any value of L.
